.github/workflows/dbt_run.yml

```
@@ -16,6 +16,30 @@ jobs:
     steps:
       - name: Check out repository code
         uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

copilot tells me this fetches all history for all branches, rather than just most recent commit. do we need that much for this to work? or can be less commits?

---

I think we can try `2` for experiments first.
```suggestion
        with:
          fetch-depth: 2
```

But maybe we need `fetch-depth: 0` instead of `fetch-depth: 2` because:

1. While `fetch-depth: 2` would be sufficient for direct pushes (HEAD~1 comparison), it could fail in PR scenarios where:
   - The PR branch diverged from main more than 2 commits ago
   - We need to fetch and compare against main branch (`git fetch origin main`)

2. `fetch-depth: 0` ensures the workflow is reliable in all scenarios by fetching the complete history.

The additional clone time from fetching full history is negligible compared to the dbt run time, and the robustness benefit outweighs this small overhead.